src/proxy/connection_manager.rs

```diff
@@ -208,13 +215,12 @@ impl ConnectionManager {
             .write()
```

Comment:
Do we even need `track` as a discrete call here, if it's only returning `Option<watch>`?

It's only ever called right after `register` (which now `tracks`) so `register` might as well return the `Option<watch>`?

Reply:
I've thought about this a bit. I'm not sure. The difference is we do assert RBAC between register and track. So we would track and immediately drop if they fail RBAC. That's probably ok though

Reply:
Yes, I think it comes out to the same thing from the perspective of the caller, and the internal state.

Reply:
Ok merged them, I think its fine